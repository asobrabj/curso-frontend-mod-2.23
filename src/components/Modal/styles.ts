import styled from 'styled-components'

interface IModalStyle {
  $visible: boolean
}

export const Container = styled.div<IModalStyle>`
  /* z-index: 1; */
  background-color: rgba(0, 0, 0, 0.4);
  position: fixed;
  top: 0;
  width: 100vw;
  height: 100vh;
  display: ${(props) => (props.$visible ? 'flex' : 'none')};
  align-items: center;
  justify-content: center;
`
