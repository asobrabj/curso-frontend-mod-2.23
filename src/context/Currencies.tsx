import { createContext, useContext, useState } from 'react';
import instance from '../service/axiosaInstace';

export interface ICurrencyContext {
  to: string;
  setTo: React.Dispatch<React.SetStateAction<string>>;
  from: string;
  setFrom: React.Dispatch<React.SetStateAction<string>>;
  amount: string;
  setAmount: React.Dispatch<React.SetStateAction<string>>;
  fecthConversion: (
    to: string,
    from: string,
    amount: string,
  ) => Promise<string>;
  error: boolean;
}

interface ICurrencyProvider {
  children: React.ReactNode;
}

const CurrencyContext = createContext<ICurrencyContext>({} as ICurrencyContext);

export const CurrencyProvider: React.FC<ICurrencyProvider> = ({ children }) => {
  const [error, setError] = useState<boolean>(false);
  const [to, setTo] = useState<string>('');
  const [from, setFrom] = useState<string>('');
  const [amount, setAmount] = useState<string>('');

  const fecthConversion = async (
    to: string,
    from: string,
    amount: string,
  ): Promise<string> => {
    try {
      const response = await instance.get(`${to}/${from}/${amount}`);
      return response.data.conversion_result;
    } catch (err) {
      setError(true);
      return 'Erro na convers√£o';
    }
  };

  return (
    <CurrencyContext.Provider
      value={{
        fecthConversion,
        error,
        amount,
        from,
        to,
        setTo,
        setAmount,
        setFrom,
      }}
    >
      {children}
    </CurrencyContext.Provider>
  );
};

export const useCurrency = () => {
  return useContext(CurrencyContext);
};
