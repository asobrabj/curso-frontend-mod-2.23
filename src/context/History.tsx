/* eslint-disable react-refresh/only-export-components */
import { createContext, useContext, useEffect, useState } from 'react';

export interface IDataCurrency {
  to: string;
  from: string;
  amount: string;
  result: string;
  date: string;
}

export interface IHistoryContext {
  setHistory: (data: IDataCurrency) => void;
  history?: Array<IDataCurrency>;
}

interface IHistoryProvider {
  children: React.ReactNode;
}

const HistoryContext = createContext<IHistoryContext>({} as IHistoryContext);

export const HistoryProvider: React.FC<IHistoryProvider> = ({ children }) => {
  const [history, setHistoryState] = useState<Array<IDataCurrency>>();

  useEffect(() => {
    const currentData = window.localStorage.getItem('history');
    const newData = currentData ? JSON.parse(currentData) : [];
    setHistoryState(newData);
  }, []);

  const setHistory = (data: IDataCurrency) => {
    const currentData = window.localStorage.getItem('history');
    const newData = currentData ? JSON.parse(currentData) : [];
    newData.push(data);
    setHistoryState(newData);
    window.localStorage.setItem('history', JSON.stringify(newData));
    return;
  };

  return (
    <HistoryContext.Provider value={{ setHistory, history }}>
      {children}
    </HistoryContext.Provider>
  );
};

export const useHistory = () => {
  return useContext(HistoryContext);
};
