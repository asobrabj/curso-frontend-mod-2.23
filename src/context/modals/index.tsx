import { createContext, useContext, useState } from 'react'
import IAnotation from '../../types/anotation'

export interface IModalContext {
  isVisible: boolean
  setIsVisible: React.Dispatch<React.SetStateAction<boolean>>
  isForm: boolean
  setIsForm: React.Dispatch<React.SetStateAction<boolean>>
  dataCard: IAnotation
  setDataCard: React.Dispatch<React.SetStateAction<IAnotation>>
  idCard: number | null
  setIdCard: React.Dispatch<React.SetStateAction<number | null>>
}

interface IModalProvider {
  children: React.ReactNode
}

const ModalContext = createContext<IModalContext>({} as IModalContext)

export function ModalProvider({ children }: IModalProvider) {
  const [isVisible, setIsVisible] = useState<boolean>(false)
  const [isForm, setIsForm] = useState<boolean>(false)
  const [idCard, setIdCard] = useState<number | null>(null)
  const [dataCard, setDataCard] = useState<IAnotation>({
    id: null,
    concluded: false,
    date: '',
    description: '',
    name: '',
    category: '',
    priority: '',
    time: '',
  })

  return (
    <ModalContext.Provider
      value={{
        idCard,
        setIdCard,
        isVisible,
        setIsVisible,
        isForm,
        setIsForm,
        dataCard,
        setDataCard,
      }}
    >
      {children}
    </ModalContext.Provider>
  )
}

export function useModal() {
  return useContext(ModalContext)
}
