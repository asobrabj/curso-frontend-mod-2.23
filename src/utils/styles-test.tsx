import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from '../styles/darkTheme'

export const renderTheme = (Children: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}> {Children} </ThemeProvider>)
}
