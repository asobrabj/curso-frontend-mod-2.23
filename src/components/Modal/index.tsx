import { useModal } from '../../context/modals'
import CardDetail from '../CardDetail'
import FormAddAnotation from '../FormAddAnnotation'
import * as S from './styles'

export default function Modal() {
  const { isVisible, isForm } = useModal()

  return (
    <S.Container $visible={isVisible}>
      {isForm ? <FormAddAnotation /> : <CardDetail />}
    </S.Container>
  )
}
