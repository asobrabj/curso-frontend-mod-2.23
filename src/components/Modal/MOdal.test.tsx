import { screen } from '@testing-library/react'
import { useModal } from '../../context/modals'
import { renderTheme } from '../../utils/styles-test'
import Modal from '../Modal'

const mockUseModal = useModal as jest.Mock
jest.mock('../../context/modals', () => ({
  useModal: jest.fn(),
}))

// Mock dos componentes CardDetail e FormAddAnotation
jest.mock('../CardDetail', () => jest.fn(() => <div>CardDetail Component</div>))
jest.mock('../FormAddAnnotation', () =>
  jest.fn(() => <div>FormAddAnnotation Component</div>)
)

describe('Modal Component', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('deve renderizar o componente CardDetail quando isVisible é true e isForm é false', () => {
    mockUseModal.mockReturnValue({ isVisible: true, isForm: false })

    renderTheme(<Modal />)

    expect(screen.getByText('CardDetail Component')).toBeInTheDocument()
    expect(
      screen.queryByText('FormAddAnnotation Component')
    ).not.toBeInTheDocument()
  })

  it('deve renderizar o componente FormAddAnnotation quando isVisible é true e isForm é true', () => {
    mockUseModal.mockReturnValue({ isVisible: true, isForm: true })

    renderTheme(<Modal />)

    expect(screen.getByText('FormAddAnnotation Component')).toBeInTheDocument()
    expect(screen.queryByText('CardDetail Component')).not.toBeInTheDocument()
  })

  it('deve renderizar null quando isVisible é false', () => {
    mockUseModal.mockReturnValue({ isVisible: false, isForm: false })

    renderTheme(<Modal />)

    expect(screen.queryByText('CardDetail Component')).toBeInTheDocument()
    expect(
      screen.queryByText('FormAddAnnotation Component')
    ).not.toBeInTheDocument()
  })
  it('snapshot', () => {
    const { container } = renderTheme(<Modal />)
    expect(container).toMatchSnapshot()
  })
})
