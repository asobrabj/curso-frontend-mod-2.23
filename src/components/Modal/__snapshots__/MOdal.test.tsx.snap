// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component snapshot 1`] = `
.c0 {
  background-color: rgba(0, 0, 0, 0.4);
  position: fixed;
  top: 0;
  width: 100vw;
  height: 100vh;
  display: none;
  align-items: center;
  justify-content: center;
}

<div>
   
  <div
    class="c0"
  >
    <div>
      CardDetail Component
    </div>
  </div>
   
</div>
`;
